packages/core/src/utils/htmlValidationUtil.ts

```diff
@@ -0,0 +1,18 @@
+import cheerio from 'cheerio';
+import * as logger from './logger';
+
+function logWarningForMissingTbody(rootNode: cheerio.Root, path: string) {
+  const tables = rootNode('table');
+  for (let i = 0; i < tables.length; i += 1) {
+    const table = rootNode(tables[i]);
+    if (table.find('tbody').length === 0) {
+      logger.error(`Invalid HTML in ${path}.\n`
+       + 'Table must have a tbody tag. Please correct this to avoid Vue hydration issues.\n');
+    }
+  }
+}
+
+export function checkForVueHydrationViolation(content: string, path: string) {
+  const $ = cheerio.load(content);
```

Comment:
I am thinking whether we can make it such that it is better to integrate more checks? Currently it feels as though for future we will do something similar and I think if we do cheerio.load for every check it is gonna take a while longer.

Is it possible to make it such that 1 cheerio.load for multiple checks in the future without refactoring the whole code.

Reply:
Yes, and that is the current design: At page/index,ts, it will call `checkForVueHydrationViolation` with content as string. The cheerio.load will only be called once per Page.

<img width="703" alt="Screenshot 2024-04-18 at 17 10 48" src="https://github.com/MarkBind/markbind/assets/121547057/df365990-1318-44b1-ba5e-9ad87a4a8378">

Reply:
Your current implementation is good then. No need to change 👍